prov/gni/include/gnix_mr_cache.h

```diff
@@ -220,10 +221,18 @@ int _gnix_mr_cache_destroy(gnix_mr_cache_t *cache);
 int _gnix_mr_cache_flush(gnix_mr_cache_t *cache);
 
 /**
- * TODO
+ * @brief Initializes the MR cache state
+ *
+ * @param[in,out] cache   a gnix memory registration cache
+ * @param[in] mr_notifier notifier, @see struct gnix_mr_notifier
+ * @param[in] attr        cache attributes, @see gnix_mr_cache_attr_t
+ *
+ * @return           FI_SUCCESS on success
+ *                   -FI_ENOMEM otherwise
  */
 int _gnix_mr_cache_init(
 		gnix_mr_cache_t      **cache,
+		struct gnix_mr_notifier *mr_notifier,
```

Comment:
There is a nice side benefit to putting this in the attributes and checking to see if it has been provided or not. If it hasn't been provided, we can either use the default notifier (stubbed and returned -FI_EBUSY) or just check to see if it exists when necessary (if (cache->notifier) do_something(event)). This should keep it pretty agnostic as well.


--Jim

________________________________
From: Sung-Eun Choi <notifications@github.com>
Sent: Monday, May 9, 2016 6:19:55 PM
To: ofi-cray/libfabric-cray
Cc: James Swaro; Mention
Subject: Re: [ofi-cray/libfabric-cray] Add kdreg notifier to memory registration cache (and clean up) (#809)


In prov/gni/include/gnix_mr_cache.h<https://github.com/ofi-cray/libfabric-cray/pull/809#discussion_r62589686>:

>   */
>  int _gnix_mr_cache_init(
>               gnix_mr_cache_t      **cache,
> +             struct gnix_mr_notifier *mr_notifier,


Good idea.

-
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/ofi-cray/libfabric-cray/pull/809/files/a98a4782dc49e4ddd84f1b37eb6f373b51800594#r62589686>